src/views/Signup.jsx

```diff
@@ -40,7 +41,7 @@ export default () => {
 
     try {
       const signUpResult = await createUserWithEmailAndPassword(email, password);
-      await signUpResult.user.sendEmailVerification();
+      await signUpResult.user.sendEmailVerification({ url: `${baseUrl}/${returnUrl}` });
```

Comment:
This is lacking the hash sign and hence does not redirect to the right location. Should be like in my suggestion.
```suggestion
      await signUpResult.user.sendEmailVerification({ url: `${baseUrl}/#/${returnUrl}` });
```